### PR TITLE
Add PyDocStyle Support

### PR DESCRIPTION
**Pull request by juanitorduz**
Give some love to our docstrings to ensure they are of the quality we want. https://docs.astral.sh/ruff/rules/#pydocstyle-d can help us with that!

~~290 Errors to fix 💪~~ ✅ 

**This PR does not change any code, just doc-strings.**

----
📚 Documentation preview 📚: https://pymc-marketing--951.org.readthedocs.build/en/951/

### CODE REVIEW

**Comment by juanitorduz** [comment]
ok just 100 more errors to go!

File: pymc_marketing/mmm/base.py [hunk 1]
```diff
@@ -867,11 +895,18 @@ def plot_channel_contribution_share_hdi(
         return fig
 
     def graphviz(self, **kwargs):
+        """Get the graphviz representation of the model.
+
+        Returns
+        -------
```
Comment (wd60622) [inline pymc_marketing/mmm/base.py:901]:
```suggestion
        """Get the graphviz representation of the model.
        
        Parameters
        ----------
        **kwargs
            Keyword arguments for the `pm.model_to_graphviz` function

        Returns
        -------
```

File: pymc_marketing/mmm/delayed_saturated_mmm.py [hunk 2]
```diff
@@ -916,17 +959,20 @@ class MMM(
     def channel_contributions_forward_pass(
         self, channel_data: npt.NDArray[np.float64]
     ) -> npt.NDArray[np.float64]:
-        """Evaluate the channel contribution for a given channel data and a fitted model, ie. the forward pass.
-        We return the contribution in the original scale of the target variable.
+        """Evaluate the channel contribution for a given channel data and a fitted model.
+
+        Hence, the forward pass.We return the contribution in the original scale of the target variable.
```
Comment (wd60622) [inline pymc_marketing/mmm/delayed_saturated_mmm.py:964]: what do you mean here?

Comment (wd60622) [reply pymc_marketing/mmm/delayed_saturated_mmm.py:964]: "the forward pass.We" doesn't make sense

**Review by wd60622 — APPROVED** [review]
Good stuff. Thanks for this! 
Just a small suggestion and comment

File: pymc_marketing/mmm/delayed_saturated_mmm.py (hunk 2 above)
Comment (juanitorduz) [reply pymc_marketing/mmm/delayed_saturated_mmm.py]: Thanks! Fixed in https://github.com/pymc-labs/pymc-marketing/pull/951/commits/0af7d7fce1e477b60e7e73105e6932bafba34d2e

**Pull request merged**